nixos/modules/services/networking/dns-over-https.nix

```diff
@@ -0,0 +1,193 @@
+  '';
+
+in
+{
+  options = {
+    services.dns-over-https = {
+      verbose = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Enable logging";
+      };
+      client = mkOption {
+        description = "DNS-over-HTTPS Client";
+        type = types.submodule {
+          options = {
+            enable = mkEnableOption "DNS-over-HTTPS Client";
+            bootstrap = mkOption {
+              description = "Bootstrap DNS server to resolve the address of the upstream resolver";
+              type = types.listOf types.str;
+              default = [ "8.8.8.8:53" "8.8.4.4:53" ];
+              example = [
+                # Google's resolver, bad ECS, good DNSSEC
+                "8.8.8.8:53"
+                "8.8.4.4:53"
+                # CloudFlare's resolver, bad ECS, good DNSSEC
+                "1.1.1.1:53"
+                "1.0.0.1:53"
+              ];
+            };
+            listen = mkOption {
+              description = "DNS listen port";
+              type = types.listOf types.str;
+              default = [
+                "127.0.0.1:53"
+                "127.0.0.1:5380"
+                "[::1]:53"
+                "[::1]:5380"
+              ];
+              example = [ ":53" ];
+            };
+            upstreamSelector = mkOption {
+              type = types.enum [ "random" "weighted_round_robin" "lvs_weighted_round_robin" ];
+              description = "upstream selection algorithm";
+              default = "random";
+            };
+            upstream = mkOption {
+              description = "list of upstream servers to query";
+              type = with types; listOf(submodule {
+                options = {
+                  url = mkOption {
+                    type = str;
+                    description = "URL for the backend, usually 'https://.../dns-query'";
+                  };
+                  weight = mkOption {
+                    type = ints.u8;
+                    description =
+                      "weight should be in (0, 100], if upstream_selector is random, weight will be ignored";
+                    default = 50;
+                  };
+                };
+              });
+              default = [ { url = "https://cloudflare-dns.com/dns-query"; } ];
+              example = [
+                # Google's resolver, good ECS, good DNSSEC
+                { url = "https://dns.google/dns-query"; }
+                # CloudFlare's resolver, bad ECS, good DNSSEC
+                { url = "https://cloudflare-dns.com/dns-query"; }
+                # CloudFlare's resolver, bad ECS, good DNSSEC
+                { url = "https://1.1.1.1/dns-query"; }
+                # DNS.SB's resolver, good ECS, good DNSSEC
+                { url = "https://doh.dns.sb/dns-query"; }
+                # Quad9's resolver, bad ECS, good DNSSEC
+                { url = "https://9.9.9.9/dns-query"; }
+                # CloudFlare's resolver for Tor, available only with Tor
```

Comment:
```suggestion
                # CloudFlare's resolver via Tor
```

Reply:
Thanks, was copied from the upstream example